### DISCUSSION TITLE
More documentation on creating new rules

### DISCUSSION THREAD

**Discussion by fabienpe** [ClearURLs/Addon#312] [General]
Hello,

Is it possible to provide more detailed documentation on creating new rules ?

For instance I've been trying to get the following URLs cleaned:
https://www.darkreading.com/risk/cybersecurity-talk-about-climate-change?_mc=NL_DR_EDT_DR_weekly_20230810&cid=NL_DR_EDT_DR_weekly_20230810&sp_aid=117669&elq_cid=43071608&sp_eh=caeccb6c48dba8703c77ef3ddb73eae2626d2da7f8e3803522e76ff66be47a83&sp_eh=caeccb6c48dba8703c77ef3ddb73eae2626d2da7f8e3803522e76ff66be47a83&sp_cid=49452

I expect the following as output
https://www.darkreading.com/risk/cybersecurity-talk-about-climate-change

No luck so far with the following.

````
"darkreading":
        {
            "urlPattern": "^https?:\\/\\/(?:[a-z0-9-]+\\.)*?darkreading\\.com",
            "rules": [
                "_mc",
                "cid",
                "sp_aid",
                "elq_cid",
                "sp_eh",
                "sp_eh",
                "sp_cid"
            ]
        }
```

**Comment by stokito** [comment]
https://docs.clearurls.xyz/1.27.3/specs/rules/
